fix eventmap handling

Ankurk99 opened the pull request.

Ankurk99 committed:
fix eventmap handling

Signed-off-by: Ankur Kothiwal <ankur.kothiwal@accuknox.com>

nam-jaehyun commented on KubeArmor/BPF/entrypoint.bpf.c:
```diff
@@ -29,9 +29,12 @@ enum {
+    uint32_t *val;
+
+    val = bpf_map_lookup_elem(&ka_ea_event_map, &key);
+    if (!val || *val == 0)
```
why did you remove the line to check if the 'val' pointer is null?

Ankurk99 replied:
Here !val is checking for the previous (val == 0) condition

nam-jaehyun commented on KubeArmor/eventAuditor/entryPointHandler.go:
```diff
@@ -132,30 +133,44 @@ func (ea *EventAuditor) DestroyEntryPointPrograms(bman *KABPFManager) error {
 func (ea *EventAuditor) EnableEntryPoint(probe string) {
 	var eventMapElem EventElement
 
+	probe = strings.Replace(probe, "sys_", "", -1)
```
if strings.HasPrefix(probe, "sys_") {
   ...
}
we need to make a room for other kprobes too.

Ankurk99 replied:
Okay, will change it

nam-jaehyun commented on KubeArmor/eventAuditor/entryPointHandler.go:
```diff
@@ -132,30 +133,44 @@ func (ea *EventAuditor) DestroyEntryPointPrograms(bman *KABPFManager) error {
+	_, supported := ea.SupportedEntryPoints[probe]
+
+	if !supported {
+		fmt.Println(probe, "is currently not supported")
```
ea.Logger.Errf("%s is currently not supported", probe)

nam-jaehyun commented on KubeArmor/eventAuditor/entryPointHandler.go:
```diff
@@ -132,30 +133,44 @@ func (ea *EventAuditor) DestroyEntryPointPrograms(bman *KABPFManager) error {
 	}
 
-	// TODO: remove it once fixing the update
+	// for debugging
```
If the logic works, we can remove 151-152.

Ankurk99 replied:
We are not getting the required output from the trace_pipe, even when the map is being correctly updated. This part was left for debugging this error.

nam-jaehyun replied:
you mean the current code is not working?

Ankurk99 replied:
yes, there is some issue with the "val" in entrypoint.bpf.c, we are not getting the required output from the trace_pipe.
The syscalls whose value is set to 1 (and are enabled), are being skipped.

nam-jaehyun commented on KubeArmor/eventAuditor/entryPointHandler.go:
```diff
@@ -132,30 +133,44 @@ func (ea *EventAuditor) DestroyEntryPointPrograms(bman *KABPFManager) error {
 func (ea *EventAuditor) DisableEntryPoint(probe string) {
 	var eventMapElem EventElement
 
+	probe = strings.Replace(probe, "sys_", "", -1)
```
the same comment with the above one.

nam-jaehyun commented on KubeArmor/eventAuditor/entryPointHandler.go:
```diff
@@ -132,30 +133,44 @@ func (ea *EventAuditor) DestroyEntryPointPrograms(bman *KABPFManager) error {
+	_, supported := ea.SupportedEntryPoints[probe]
+
+	if !supported {
+		fmt.Println(probe, "is currently not supported")
```
the same comment with the above one.

nam-jaehyun commented on KubeArmor/eventAuditor/entryPointHandler.go:
```diff
@@ -132,30 +133,44 @@ func (ea *EventAuditor) DestroyEntryPointPrograms(bman *KABPFManager) error {
 	}
 
-	// TODO: remove it once fixing the update
+	// for debugging
```
the same comment with the above one.

nam-jaehyun commented on KubeArmor/eventAuditor/entryPointHandler.go:
```diff
@@ -172,6 +187,7 @@ func (ea *EventAuditor) UpdateEntryPoints(auditPolicies *map[string]tp.AuditPoli
 	// all entrypoints list
 	for _, policy := range AuditPolicies {
 		for _, event := range policy.Events {
+			event.Probe = strings.Replace(event.Probe, "sys_", "", -1)
```
the same comment with the above one.

nam-jaehyun commented on KubeArmor/eventAuditor/entryPointHandler.go:
```diff
@@ -182,6 +198,7 @@ func (ea *EventAuditor) UpdateEntryPoints(auditPolicies *map[string]tp.AuditPoli
 
 	// new entrypoints to be attached
 	for _, newProbe := range entrypointList {
+		newProbe = strings.Replace(newProbe, "sys_", "", -1)
```
the same comment with the above one.

nam-jaehyun commented on KubeArmor/eventAuditor/entryPointHandler.go:
```diff
@@ -196,6 +213,7 @@ func (ea *EventAuditor) UpdateEntryPoints(auditPolicies *map[string]tp.AuditPoli
 
 	// old entrypoints to be detached
 	for _, oldProbe := range ea.ActiveEntryPoints {
+		oldProbe = strings.Replace(oldProbe, "sys_", "", -1)
```
the same comment with the above one.

Ankurk99 committed:
fix ea_event map to update correctly

The map was not being currectly updated due to the presence of the
duplicate file ka_ea_event_map.bpf.c

Signed-off-by: Ankur Kothiwal <ankur.kothiwal@accuknox.com>
